packages/flutter_tools/lib/src/android/android_device.dart

```diff
@@ -615,13 +625,18 @@ class AndroidDevice extends Device {
 
   static final RegExp _timeRegExp = RegExp(r'^\d{2}-\d{2} \d{2}:\d{2}:\d{2}\.\d{3}', multiLine: true);
 
-  /// Return the most recent timestamp in the Android log or null if there is
+  /// Return the most recent timestamp in the Android log or [null] if there is
   /// no available timestamp. The format can be passed to logcat's -T option.
   String get lastLogcatTimestamp {
-    final String output = runAdbCheckedSync(<String>[
-      'shell', '-x', 'logcat', '-v', 'time', '-t', '1',
-    ]);
-
+    String output;
+    try {
+      output = runAdbCheckedSync(<String>[
+        'shell', '-x', 'logcat', '-v', 'time', '-t', '1',
+      ]);
+    } catch (error) {
```

Comment:
This should also catch `ProcessException` only, but that likely requires modifying `_runWithLoggingSync` in process.dart to throw a `ProcessException` instead of a `String`.

Reply:
I filed #36242. What about we change exception type in a separate PR?

Reply:
Sure, if you think that will make that change easier.